examples/micropython/Makefile

```diff
@@ -0,0 +1,42 @@
+# name of your application
+APPLICATION = micropython
+
+# If no BOARD is found in the environment, use this default:
+BOARD ?= native
+
+# This has to be the absolute path to the RIOT base directory:
+RIOTBASE ?= $(CURDIR)/../..
+
+# the MicroPython port currently only supports 32bit architectures
+BOARD_BLACKLIST := arduino-duemilanove arduino-leonardo \
+                   arduino-mega2560 arduino-nano arduino-uno chronos \
+                   jiminy-mega256rfr2 mega-xplained msb-430 msb-430h \
+                   telosb waspmote-pro wsn430-v1_3b wsn430-v1_4 z1
+
+# the MicroPython port depends on ISR_STACKSIZE, which is not available on
+# arm7, risc-v, pic32
+BOARD_BLACKLIST += avsextrem msba2
+BOARD_BLACKLIST += hifive1
+BOARD_BLACKLIST += pic32-clicker pic32-wifire
+
+# boards with too little RAM or flash for MicroPython
+BOARD_INSUFFICIENT_MEMORY := blackpill bluepill i-nucleo-lrwan1 nucleo-f030r8 \
+                             nucleo-f031k6 nucleo-f042k6 nucleo-f302r8 \
+                             nucleo-f303k8 nucleo-f334r8 nucleo-l031k6 \
+                             nucleo-l053r8 opencm904 saml10-xpro saml11-xpro \
+                             stm32f0discovery spark-core stm32l0538-disco
+
+# select MicroPython package
+USEPKG += micropython
+
+# include blop.py as header
+BLOBS += boot.py
+
+# configure MicroPython's heap size
+CFLAGS += '-DMP_RIOT_HEAPSIZE=(8U*1024)'
```

Comment:
Could be moved in the package `Makefile.include` file. This is not specific to the application, is it ?

Reply:
it is here to show that it can be configured. if left out, there's a default as fallback.

Reply:
IMHO, the best solution is to document it in the package documentation.

Reply:
done

Reply:
I'm not happy with the proposed solution: too low level, not well documented.

May I suggest to add the following in Makefile.include?
```mk
MP_RIOT_HEAPSIZE ?= 8182U

CFLAGS += -DMP_RIOT_HEAPSIZE=$(MP_RIOT_HEAPSIZE)
```
 (I don't think the quotes are required btw)

And in doc.txt:
```md
## Configuration options
 *
 * Use the following environment variables in the application Makefile
 * or from the command line to configure MicroPython:
 *
 * MP_RIOT_HEAPSIZE: heap size for Micropython. Defaults to 8KiB.
 *
 * Example on the command line:
 * ```
 * MP_RIOT_HEAPSIZE=16 make -C examples/micropython
 * ```
```

btw, the default is 8KiB, not 16.

Reply:
done